File: Editor/src/main/java/com/rspsi/studio/StudioBuildInfo.java
package com.rspsi.studio;

/** Lightweight build metadata safe to use before any project/cache is opened. */
public final class StudioBuildInfo {
    private StudioBuildInfo() {}

    public static String version() {
        String explicit = System.getProperty("openrune.studio.version");
        if (explicit != null && !explicit.isBlank()) return explicit;

        Package pkg = StudioBuildInfo.class.getPackage();
        String implementation = pkg == null ? null : pkg.getImplementationVersion();
        return implementation == null || implementation.isBlank() ? "dev" : implementation;
    }

    public static String displayVersion() {
        return "OpenRune Content Studio  v" + version();
    }
}

File: Editor/src/main/kotlin/com/rspsi/studio/StudioBuildInfo.kt
package com.rspsi.studio

/** Lightweight build metadata safe to use before any project/cache is opened. */
object StudioBuildInfo {
    private const val VERSION_PROPERTY = "openrune.studio.version"
    private const val DEVELOPMENT_VERSION = "dev"

    @JvmStatic
    fun version(): String {
        System.getProperty(VERSION_PROPERTY)
            ?.takeIf { it.isNotBlank() }
            ?.let { return it }

        return StudioBuildInfo::class.java.package
            ?.implementationVersion
            ?.takeIf { it.isNotBlank() }
            ?: DEVELOPMENT_VERSION
    }

    @JvmStatic
    fun displayVersion(): String = "OpenRune Content Studio  v${version()}"
}
